content/ui-ux.md

```diff
@@ -0,0 +1,514 @@
+
+  - Your collections, as well as the `Meteor.users` collection,
+  - Accounts information, like `Meteor.user()` and `Meteor.loggingIn()`
+  - Current route information
+  - Any other client-side data stores (see XXX not sure which article? maybe this one)
+
+### Smart Components
+
+Of course sometimes you do need to access the global data sources to feed data into your application. We call components that need to access data "smart". Such components typically the following things
+
+ 1. Subscribe to data, using subscriptions.
+ 2. Fetch data from those subscriptions.
+ 3. Fetch global client-side state from stores such as the Router, Accounts, and your own stores.
+
+Ideally, once a smart component has assembled such a set of data, it passes it off to a pure component child to render with. The smart component actually does not render anything apart from one or more pure children.
+
+A typical use case for a smart component is the "page" component that the router points you to when you access a URL. Such a component typically needs to do the three things above and then can pass the resulting arguments into child components. In the Todos example app, the `listShowPage` does exactly this, with a resultingly simple template:
+
+```
+<template name="listsShowPage">
+  {{#each listIdArray}}
+    {{> listsShow todosReady=Template.subscriptionsReady list=list}}
+  {{/each}}
+</template>
+```
+
+## Visually testing pure components
+
+A useful property of pure components is that you can render them anywhere because they don't rely on complicated environments. One very useful thing that this enables is a component _styleguide_ or harness.
+
+A styleguide consists of two parts:
+
+1. A list of "entries"---a component coupled with a list of specifications; basically a list of different arguments that the component expects which trigger different behaviour.
+
+2. A special route in the development version of the application that renders one or more components with one or more of the specificiations.
+
+For instance, in Galaxy, we have a component styleguide that renders each pure component either one specification at a time, or with all specifications at once. 
+
+[ss]
+
+Such rendering enables very quick development of visual aspects of the component in all possible states. Typically in a complex application, it can be quite difficult to achieve certain states of components purely by "using" the application. For example, in Galaxy, the component screen enters quite a complex state if two deploys to the same application are happening simutaneously. This is a very hard to state to reach for a extended period of time if you are simply using the application!
+
+[ss]
+
+XXX: could say a lot more about this, but I feel like we might just blog about it / do a whole article on it's own once we've had a chance to release some code.
```

Comment:
I'd say in the meantime, we should like to some other blog post about this - I'm sure there are many since a lot of companies seem to be switching to this approach.